scripts/login-page.nse

```diff
@@ -0,0 +1,155 @@
+local http = require "http"
+local io = require "io"
+local nmap = require "nmap"
+local shortport = require "shortport"
+local stdnse = require "stdnse"
+local string = require "string"
+
+description = [[
+Exposes the admin login page in any website.
+Displays both the user login and admin login pages in any website.
+
+TODO:
+- Automatically crawl the website and find the extension instead of taking
+  it as a parameter from the user.
+    - httpspider library can be used to accomplish this task.
+- If there are frequent socket errors or http.request TIMEOUTS notify the user
+  to check his internet connection and proxy instead of returning nil.
+- Current database are having only 150 entries for each extension, updating this
+  is a never ending process and this can be done frequently to obtain better results.
+]]
+
+---
+--  @usage ./nmap --script login-page <target> -d
+--  @usage ./nmap --script login-page --script-args extension="php" <target> -d
+--
+--  If timeout occurs frequently due to bad internet connection then
+--  @usage ./nmap --script login-page --script-args extension="php" <target> --host-timeout=<timeout> -d
+--
+--  Best way to run the script
+--  If the user has prior knowledge on which port to check, he can save time by
+--  specifying that particular port as a general command line argument using -p
+--  @usage ./nmap --script login-page --script-args extension="jsp" <target> -p 80 -d
+--
+--  @args login-page.extension Checks for pages of particular extension,
+--        default is extension is all which checks for all the extensions.
+--
+--  @output
+--  PORT   STATE SERVICE REASON
+--  22/tcp open  ssh     syn-ack ttl 64
+--  80/tcp open  http    syn-ack ttl 64
+--  | login-page:
+--  |   192.168.146.145/admin/
+--  |   192.168.146.145/admin/index.php
+--  |_  192.168.146.145/admin/login.php
+---
+
+author = "Rewanth Cool"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"discovery", "safe"}
+
+portrule = function(host, port)

```

Comment:
@h4ck3rk3y I didn't use short.http because it doesn't work when the user mention specific ports through command line. For ex, if I want to run this script on port 8090 **only**, then shortport.http doesn't work when I pass the 8090 using -p option.

I customized portrule function in such a way that even if the user wants to run this script against port 22 he can run it using -p 22 from command line.

If my portrule function is used with the following command,
`./nmap --script login-page 192.168.146.145 --script-args extension="php" -p 22 -d`
It sends GET request to http://192.168.146.145:22/admin/cp.php

But, if I use shortport.http instead of my function with the following command,
`./nmap --script login-page 192.168.146.145 --script-args extension="php" -p 22 -d`
It fails to send the GET request to http://192.168.146.145:22/admin/cp.php and instead it quits immediately.

I know that http is not running on port 22 but still if the user wants to run the script against that port, I thought this function might be helpful.

Correct me if I'm wrong.